### PR TITLE
Port ability to get physics diagnostics

### PR DESCRIPTION
**Pull request by spencerkclark**
This PR ports the ability to get physics diagnostics from SHiELD from https://github.com/ai2cm/fv3gfs-wrapper/pull/161.  In SHiELD these come from `FV3GFS_io.F90` instead of `GFS_diagnostics.F90`.  This PR relies on changes recently merged in SHiELD in https://github.com/NOAA-GFDL/SHiELD_physics/pull/30; the SHiELD_physics submodule is updated accordingly.

### CODE REVIEW

File: wrapper/tests/test_diagnostics.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            self.assertIsInstance(item.module_name, str)
+
+            self.assertIsInstance(item.name, str)
+            self.assertIsNot(item.name, "")
+
+            self.assertIsInstance(item.description, str)
+            self.assertIsInstance(item.unit, str)
+
+    def test_get_diagnostic_data(self):
+        names_to_get = ["dt3dt_1", "u10m"]
```
Comment (spencerkclark) [inline wrapper/tests/test_diagnostics.py:37]: While getting the accumulated diagnostic `"dt3dt_1"` is fine for testing, it is probably in better taste to illustrate getting a non-accumulated 3D diagnostic:
```suggestion
        names_to_get = ["dkt_pbl", "u10m"]
```

File: wrapper/templates/_wrapper.pyx [hunk 2]
```diff
@@ -468,3 +472,71 @@ def save_fortran_restart():
...
+
+cdef _get_diagnostic_info_by_index(int i):
+    cdef int ax
+    cdef int axes[1]
+    cdef char name[128]
+    cdef char mod_name[128]
+    cdef char desc[128]
+    cdef char unit[128]
+
+    get_metadata_diagnostics(&i, axes, &mod_name[0], &name[0], &desc[0], &unit[0])
```
Comment (brianhenn) [inline wrapper/templates/_wrapper.pyx:489]: I know this is cut from the other repo but curious why `axes` is passed by value (I think) and the rest of the args are passed by reference?

**Review by brianhenn — APPROVED** [review]
Thanks @spencerkclark looks good.

File: wrapper/tests/test_diagnostics.py (hunk 1 above)
Comment (spencerkclark) [reply wrapper/tests/test_diagnostics.py]: This accidentally got lost in the rebase, so I re-made this change in 5bdff97.

File: wrapper/tests/test_diagnostics.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            )
+            self.assertIsInstance(quantity, pace.util.Quantity)
+            assert quantity.view[:].ndim == info.axes
+            assert quantity.units == info.unit
+            if name == "eta_shal":
+                assert quantity.view[:].ndim == 3
+            elif name == "u10m":
+                assert quantity.view[:].ndim == 2
+            else:
+                raise ValueError(f"Testing only implemented for eta_shal and u10m")
```
Comment (spencerkclark) [inline wrapper/tests/test_diagnostics.py:53]:
```suggestion
                raise ValueError("Testing only implemented for eta_shal and u10m")
```

File: wrapper/templates/_wrapper.pyx [hunk 4]
```diff
@@ -468,3 +472,71 @@ def save_fortran_restart():
 def cleanup():
     """Call the Fortran cleanup routines, which clear memory and write final restart files."""
     cleanup_subroutine()
+
+
+DiagnosticInfo = namedtuple(
+    'DiagnosticInfo', 
+    ['axes', 'module_name', 'name', 'description', 'unit']
+)
```
Comment (spencerkclark) [inline wrapper/templates/_wrapper.pyx:480]:
```suggestion
DiagnosticInfo = namedtuple(
    "DiagnosticInfo", ["axes", "module_name", "name", "description", "unit"]
)
```

**Pull request merged**